### DISCUSSION TITLE
Display errors for iterated field

### DISCUSSION THREAD

**Discussion by j0hannr** [react-hook-form/react-hook-form#6894] [Q&A] [answered]
Hi guys, 

I am currently building a form where I am iterating multiple fields.
As the user will be guided through the form, I want to validate each section at a time.

On Validation with `trigger` its throwing the error:
```error
errors.apartment[i] is undefined
```
**I have isolated the error here:**
[Codesandbox](https://codesandbox.io/s/react-hook-form-js-forked-vr8o3?file=/src/App.js)

Happy for any suggestions! :)

**Comment by lundn** [comment, marked as answer]
Not react-hook-form related error I believe.

I guess at some point in the application lifecycle you're unsafely accessing nested properties on an object, which may or may not be undefined. Could you use optional chaining when accessing the errors, e.g:
`errors?.apartment?.[i]?.heater?.type`

**Reply by j0hannr** [reply]
Thats it! Thanks @lundn 👍

**Reply by bluebill1049** [reply]
thanks @lundn
